include/s3select.h

```diff
@@ -732,29 +730,29 @@ void push_function_expr::operator()(s3select* self, const char* a, const char* b
 void push_compare_operator::operator()(s3select* self, const char* a, const char* b) const
 {
   std::string token(a, b);
-  arithmetic_operand::cmp_t c = arithmetic_operand::cmp_t::NA;
+  arithmetic_operand::cmp_t c;
 
-  if (token.compare("==") == 0)
+  if (token =="=="s)
```

Comment:
as discussed , will use the operator== , not need for s.

Reply:
I have verified (https://godbolt.org/z/r6a858) and the generated code seems identical. So no problem ('though I prefer
clarifying the fact that we are comparing std::string s.